### DISCUSSION TITLE
Custom masks via API

### DISCUSSION THREAD

**Discussion by iomarcovalente** [light-and-ray/sd-webui-replacer#55] [Q&A] [answered]
I couldn't find the option in the available API so I guess that isn't available. Any chance to provide that some other way?

**Comment by light-and-ray** [comment]
Why

**Reply by iomarcovalente** [reply]
So I can add a fixed mask to the inferred one for a given picture, programmatically. But I appreciate that might not be possible

**Reply by iomarcovalente** [reply]
I might investigate if someone is exposing it as an extension for the inpaint API directly instead

**Reply by light-and-ray** [reply, marked as answer]
I mean you can use img2img api instead. I added custom mask to take advantages of hires fix and video tab. E.g removing watermarks or frames in video. Maybe only hires fix is reasonable to use in api with custom mask, but you can do the same with 2 img2img requests

**Reply by iomarcovalente** [reply]
yes thanks for the reply. I have just found that I could possibly leverage [this](https://github.com/mix1009/sdwebuiapi?tab=readme-ov-file#extension-support---segmentanything-contributed-by-timnekk) and [this](https://github.com/mix1009/sdwebuiapi?tab=readme-ov-file#img2img-inpainting)
thanks for the suggestions!
